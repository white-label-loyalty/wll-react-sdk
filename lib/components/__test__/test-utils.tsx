import { render } from '@testing-library/react';
import React from 'react';
import { CoreApiUrl, WllSdkProvider } from '../../context/WllSdkContext';
import { defaultTheme } from '../../utils/styling';

const defaultConfig = {
  apiKey: 'test-api-key',
  userToken: 'test-user-token',
  coreApiUrl: CoreApiUrl.STAGING_EU,
};

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <WllSdkProvider
      config={defaultConfig}
      theme={defaultTheme}
      navigationConfig={{}}
    >
      {children}
    </WllSdkProvider>
  );
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
