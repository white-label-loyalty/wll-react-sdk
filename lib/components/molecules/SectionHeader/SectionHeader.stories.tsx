import { Meta, StoryFn } from '@storybook/react';
import React from 'react';
import SectionHeader from './index';

export default {
  title: 'components/molecules/SectionHeader',
  component: SectionHeader,
  tags: ['autodocs'],
} as Meta;

const Template: StoryFn<typeof SectionHeader> = (args) => (
  <SectionHeader {...args} />
);

export const Default = Template.bind({});
Default.args = {
  title: 'Exclusive Offers',
  description: 'Check out our latest promotions!',
};
