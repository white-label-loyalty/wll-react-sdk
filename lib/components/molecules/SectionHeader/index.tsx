import React from 'react';
import { StyleSheet, View } from 'react-native';
import { useWllSdk } from '../../../context/WllSdkContext';
import { useResponsive } from '../../../context/ResponsiveContext';
import { getResponsiveValue } from '../../../utils/responsiveHelper';
import Text from '../../atoms/Text';

type SectionHeaderProps = {
  title?: string;
  description?: string;
};

const SectionHeader: React.FC<SectionHeaderProps> = ({
  title,
  description,
}) => {
  const { theme } = useWllSdk();
  const { isDesktop, isTablet } = useResponsive();

  if (!title) {
    return null;
  }

  const dynamicStyles = StyleSheet.create({
    sectionHeader: {
      marginBottom: getResponsiveValue(16, 8, isDesktop, isTablet),
    },
    sectionTitle: {
      fontSize: getResponsiveValue(32, 18, isDesktop, isTablet),
      marginBottom: getResponsiveValue(8, 4, isDesktop, isTablet),
    },
    sectionDescription: {
      fontSize: getResponsiveValue(24, 14, isDesktop, isTablet),
    },
  });

  return (
    <View style={dynamicStyles.sectionHeader}>
      {title && (
        <Text
          style={[
            dynamicStyles.sectionTitle,
            {
              fontWeight: '700',
              color: theme.text,
            },
          ]}
        >
          {title}
        </Text>
      )}
      {description && (
        <Text
          style={[
            dynamicStyles.sectionDescription,
            {
              color: theme.alphaDerivedText[80],
            },
          ]}
        >
          {description}
        </Text>
      )}
    </View>
  );
};

export default SectionHeader;
