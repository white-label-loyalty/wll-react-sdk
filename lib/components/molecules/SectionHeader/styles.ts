import { StyleSheet } from 'react-native';
import { useWllSdk } from '../../../context/WllSdkContext';
import { useResponsive } from '../../../hooks/useResponsive';
import { getResponsiveValue } from '../../../utils/responsiveHelper';

export const useSectionHeaderStyles = () => {
  const { isDesktop, isTablet } = useResponsive();
  const { theme } = useWllSdk();

  return StyleSheet.create({
    sectionHeader: {
      marginBottom: getResponsiveValue(16, 8, isDesktop, isTablet),
    },
    sectionTitle: {
      fontSize: getResponsiveValue(32, 18, isDesktop, isTablet),
      marginBottom: getResponsiveValue(8, 4, isDesktop, isTablet),
      fontWeight: '700',
      color: theme.text,
    },
    sectionDescription: {
      fontSize: getResponsiveValue(24, 14, isDesktop, isTablet),
      color: theme.alphaDerivedText[80],
    },
  });
};
