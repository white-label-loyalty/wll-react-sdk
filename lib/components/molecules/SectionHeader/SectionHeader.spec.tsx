import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import SectionHeader from '.';
import { render } from '../../__test__/test-utils';

describe('SectionHeader', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <SectionHeader title="Test" description="This is a test" />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with title and description', () => {
    render(<SectionHeader title="Test" description="This is a test" />);
    expect(screen.getByText('Test')).toBeInTheDocument();
    expect(screen.getByText('This is a test')).toBeInTheDocument();
  });

  it('renders only title when no description is provided', () => {
    render(<SectionHeader title="Test Title" />);
    expect(screen.getByText('Test Title')).toBeInTheDocument();
    expect(screen.queryByText('This is a test')).not.toBeInTheDocument();
  });

  it('returns null when no title is provided', () => {
    const { container } = render(
      <SectionHeader description="Description only" />
    );
    expect(container).toBeEmptyDOMElement();
  });
});
