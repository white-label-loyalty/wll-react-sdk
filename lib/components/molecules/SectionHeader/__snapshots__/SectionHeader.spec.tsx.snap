// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionHeader /> matches snapshot 1`] = `
<div>
  <div
    class="css-view-175oi2r r-marginBottom-ku1wi2"
  >
    <div
      class="css-text-146c3p1 r-fontFamily-lhg2x6 r-fontSize-1x35g6 r-marginBottom-5oul0u"
      dir="auto"
      style="font-weight: 700; color: rgb(0, 0, 0);"
    >
      Test
    </div>
    <div
      class="css-text-146c3p1 r-fontFamily-lhg2x6 r-fontSize-1i10wst"
      dir="auto"
      style="color: rgba(0, 0, 0, 0.8);"
    >
      This is a test
    </div>
  </div>
</div>
`;
